### DISCUSSION TITLE
[Question] How to do sustainable fishing?

### DISCUSSION THREAD

**Discussion by juj** [nicymike/serflings#80] [Q&A] [answered]
Reading the manual, there is a mention that one should not build too many fishing huts next to a lake:

![image](https://user-images.githubusercontent.com/225351/179930606-70256486-b1c3-4a06-b885-3cd9e54c171c.png)

Are there any rules of thumb as to how many fishermen is ok for any give size of a lake? E.g. possible to count # hexes size of the lake and do x huts per y tiles of lake? Does the game indicate in any way that a lake has been fished out?

(I hope gameplay strategy/rules related questions are ok in the issue tracker?)

**Comment by nicymike** [comment, marked as answer]
The game keeps track of the number of fish inside a lake but for you as a human player it is not possible to find out the number of fish left. The computer does check the number of fish and will burn the fisherman if no more fish is available.
I plan to add a message or to display the percentage of success to the fisherman to indicate whether the lake is empty (similar to mines). 

In general fish will move or duplicate every now and then. I think better just place one or two fisherman and just leave them alone.

**My advises**

- Too small for fisherman
 ![image](https://user-images.githubusercontent.com/19415000/179936578-04a76c37-fe71-43ac-a7bd-f75a5b14a287.png)

- One fisherman is enough
 ![image](https://user-images.githubusercontent.com/19415000/179937384-c9348234-3bd6-41a6-b034-531cc1062803.png)

- Maybe two
 ![image](https://user-images.githubusercontent.com/19415000/179937532-9a2f7fcf-3dc9-43f0-8038-6eefeb4f1ef8.png)

- Two or three are fine
![image](https://user-images.githubusercontent.com/19415000/179937665-3d30bc7f-043d-4998-b1ae-0378501e9ee0.png)

**Reply by juj** [reply]
Thanks, super helpful! This helps calibrate fishermen.

> The game keeps track of the number of fish inside a lake

> In general fish will **move** or duplicate every now and then.

Are the fishes even tracked to reside on a specific squares in the lake? And they move around it and that results in the fisher to either succeed or fail his fishing trips by chance?

If so, it would be interesting to have a debug/cheat/sandbox/gameplay option to draw all the fishes in each lake in their locations. (or draw all the fishes on a lake once one has completed a fisherman's hut on that lake)

That might make it an interesting minigame to observe what the fish population is like and how to balance sustainable fishing ("should I put one more fisherman's hut? looks like there are fewer fish now, should I remove the fisherman?" etc.)

**Reply by juj** [reply]
> whether the lake is empty (similar to mines).

When progressing time at 64x speed, I did observe that the mines get depleted. It seemed as if the miner was still happily consuming food, while producing nothing. 

That gave me a thought that it would be nice to have an option for the mines to autoclose (autoraze themselves) once they are depleted? Maybe a fisherman on a dead lake might also autoraze itself?

These kinds of changes might naturally want to get some kind of option so that they would not be enabled when/if player is searching for the original gameplay experience - but for an improved gameplay usability, autorazing might be an interesting cue to start looking for more resources elsewhere.

**Reply by nicymike** [reply]
The success for fishing depends on the number of fish next to the fisherman and a random number.

Yes, the fishes are beeing tracked. Every position with water has a number of fish assigned. The same is true for mountains and threir resources.
There is a debug key available (F3) which for example enables the display of the working area for buildings. Displaying resources (like fish) is more like a cheat in my opinion so I think there should be a separate option for this to prevent an advantage in network games or missions. You may create an issue for adding this feature.

Mines (and also most of the other buildings gathering stuff) work on pure random only. This means if a mine is empty, there may be still some resources out there but the mine worker had bad luck finding it. If you ever wondered about a lumberjack not chopping this single tree for hours, this is why. :)
Before adding something like auto raze I need to think about the random thing first. You may create an issue for this feature, too.
